mne/epochs.py

```diff
@@ -1355,16 +1359,30 @@ def _get_data(self, out=True, picks=None, item=None, verbose=None):
             logger.info('Loading data for %s events and %s original time '
                         'points ...' % (n_events, len(self._raw_times)))
 
+        orig_picks = picks
+        if orig_picks is None:
+            picks = _picks_to_idx(self.info, picks, "all", exclude=())
+        else:
+            picks = _picks_to_idx(self.info, picks)
+
+        # handle units param only if we are going to return data (out==True)
+        # else, we don't scale (scaling is vector of ones)
+        ch_factors = np.ones(len(picks))
+        if (units is not None) and out:
+            ch_factors = _get_ch_factors(self, units, picks)
+
         if self._bad_dropped:
             if not out:
                 return
             if self.preload:
                 data = data[select]
-                if picks is None:
+                if orig_picks is None:
+                    data *= ch_factors[np.newaxis, :, np.newaxis]
                     return data
                 else:
-                    picks = _picks_to_idx(self.info, picks)
-                    return data[:, picks]
+                    data = data[:, picks] * ch_factors[np.newaxis, :,
+                                                       np.newaxis]
```

Comment:
Avoid copies in memory, let broadcasting simplify things, and make slightly more computationally efficient when we can (which will be most of the time)
```suggestion
                    data = data[:, picks]
                    if units is not None:
                        data *= ch_factors[np.newaxis]
```